### PR TITLE
Upgrade examples to use TSP 4+

### LINKED ISSUE #81: Update examples to include stand-alone logdriver
Now that logdriver is a separate container, instead of being included in the TSP container, we should update version of TSP to 4.0 and include instructions for getting the logdriver container and running it as a sidecar alongside TSP using `docker compose`.

### PR DESCRIPTION
**Pull request by clintfred**
closes #81

### CODE REVIEW

File: examples/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+version: "3.3"
+services:
+  tenant-security-proxy:
+    env_file:
+      - ./demo-tsp.conf
+    ports:
+      - 32804:7777
+      - 32805:9000
+    image: gcr.io/ironcore-images/tenant-security-proxy:4.0
```
Comment (giarc3) [inline examples/docker-compose.yml:9]: Would it be worth making this just pull the `4` tag? Shouldn't be any breaking changes and could help us update this file less frequently. Just a thought

Comment (clintfred) [reply examples/docker-compose.yml]: Yep. Done.

**Review by coltfred — APPROVED** [review]
Seems like a good step for the examples. I noticed that the benchmarks are still on tsp 3.x. Should we just use the same docker compose and update those instructions as well?

**Review by BobWall23 — APPROVED** [review]

File: examples/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    image: gcr.io/ironcore-images/tenant-security-proxy:4
+    links:
+      - tenant-security-logdriver
+  tenant-security-logdriver:
+    environment:
+      - LOGDRIVER_EVENT_PRODUCER_URL=tcp://tenant-security-proxy:5555
+    env_file:
+      - ./demo-tsp.conf
+    ports:
+      - "32806:9001"
```
[the remark this answers is not in the document]
Comment (clintfred) [reply examples/docker-compose.yml:18]: No. I'll fix it.

**Review by coltfred — COMMENTED** [review]
It seems like copy/paste of the docker-compose might lead to us mistakenly updating one but not the other in the future. I feel like it might be better to just use the docker-compose.yml from examples in our benchmark code blocks. Is there a reason you didn't do it that way?

**Comment by clintfred** [comment]
demo-tsp.conf is also duplicated between the two, so I don't think it's a big deal to duplicate the docker-compose.yml.

**Pull request merged**